kafka/content.md

```diff
@@ -0,0 +1,269 @@
+
+Kafka was originally developed at LinkedIn, was open sourced in 2011, and became an Apache Software Foundation project in 2012. It is used by thousands of organizations globally to power mission-critical real-time applications, from stock exchanges, to e-commerce applications, to IoT monitoring & analytics, to name a few.
+
+%%LOGO%%
+
+## Quick start
+
+Start a Kafka broker:
+
+```noformat
+docker run -d --name broker kafka:latest
+```
+
+Open a shell in the broker container:
+
+```noformat
+docker exec --workdir /opt/kafka/bin/ -it broker sh
+```
+
+A *topic* is a logical grouping of events in Kafka. From inside the container, create a topic called `test-topic`:
+
+```noformat
+./kafka-topics.sh --bootstrap-server localhost:9092 --create --topic test-topic
+```
+
+Write two string events into the `test-topic` topic using the console producer that ships with Kafka:
+
+	./kafka-console-producer.sh --bootstrap-server localhost:9092 --topic test-topic
+
+This command will wait for input at a `>` prompt. Enter `hello`, press `Enter`, then `world`, and press `Enter again`. Enter `Ctrl+C` to exit the console producer.
+
+Now read the events in the `test-topic` topic from the beginning of the log:
+
+	./kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic test-topic --from-beginning
+
+You will see the two strings that you previously produced:
+
+	hello
+	world
+
+The consumer will continue to run until you exit out of it by entering `Ctrl+C`.
+
+When you are finished, stop and remove the container by running the following command on your host machine:
+
+	docker rm -f broker
+
+## Overriding the default broker configuration
+
+Apache Kafka supports a broad set of broker configurations that you may override via environment variables. The environment variables must begin with `KAFKA_`, and any dots in broker configurations should be specified as underscores in the corresponding environment variable. For example, to set the default number of partitions in topics, [`num.partitions`](https://kafka.apache.org/documentation/#brokerconfigs_num.partitions), set the environment variable `KAFKA_NUM_PARTITIONS`. See [here](https://github.com/apache/kafka/blob/trunk/docker/examples/README.md) for more information on overriding broker configuration in Docker.
+
+It's important to note that, if you are overriding *any* configuration, then *none* of the default configurations will be used. For example, to run Kafka in KRaft [combined mode](https://kafka.apache.org/documentation/#kraft_role) (meaning that the broker handling client requests and the controller handling cluster coordination both run in the same container) and set the default number of topic partitions to 3 instead of the default 1, we would specify `KAFKA_NUM_PARTITIONS` in addition to other required configurations:
```

Comment:
```suggestion
It's important to note that if you are overriding *any* configuration, then *none* of the default configurations will be used. For example, to run Kafka in KRaft [combined mode](https://kafka.apache.org/documentation/#kraft_role) (meaning that the broker handling client requests and the controller handling cluster coordination both run in the same container) and set the default number of topic partitions to 3 instead of the default 1, we would specify `KAFKA_NUM_PARTITIONS` in addition to other required configurations:
```